app/containers/message/Image.tsx

```diff
@@ -44,56 +57,154 @@ const Button = React.memo(({ children, onPress, disabled, theme }: IMessageButto
 		const { baseUrl, user } = useContext(MessageContext);
 		const img = imageUrl || formatAttachmentUrl(file.image_url, user.id, user.token, baseUrl);
+		const filePath = useRef('');
+
+		useLayoutEffect(() => {
+			const handleAutoDownload = async () => {
+				if (img) {
+					const searchImageCached = await searchMediaFileAsync({
+						type: MediaTypes.image,
+						mimeType: newFile.image_type,
+						messageId
+					});
+					filePath.current = searchImageCached.filePath;
+					if (searchImageCached.file?.exists) {
+						setNewFile(prev => ({
+							...prev,
+							title_link: searchImageCached.file?.uri
+						}));
+						return setToDownload(false);
+					}
+
+					if (isDownloadActive(MediaTypes.image, messageId)) {
+						return setLoading(true);
+					}
+
+					const autoDownload = await isAutoDownloadEnabled('imagesPreferenceDownload', { user, author });
+					if (autoDownload) {
+						await handleDownload();
+					}
+				}
+			};
+			handleAutoDownload();
+		}, []);
 
 		if (!img) {
 			return null;
 		}
 
+		const handleDownload = async () => {
+			setLoading(true);
+			try {
+				// The param file.title_link is the one that point to image with best quality, however we still need to test the imageUrl
+				// And we don't have sure that ever exists the file.title_link
+				const imgUrl = imageUrl || formatAttachmentUrl(newFile.title_link || newFile.image_url, user.id, user.token, baseUrl);
+				const imageUri = await downloadMediaFile({
+					downloadUrl: imgUrl,
+					mediaType: MediaTypes.image,
+					messageId,
+					path: filePath.current
+				});
+
+				setNewFile(prev => ({
+					...prev,
+					title_link: imageUri
+				}));
+				setToDownload(false);
+				setLoading(false);
+			} catch (e) {
+				setLoading(false);
+				return setToDownload(true);
+			}
+		};
+
 		const onPress = () => {
+			if (loading && isDownloadActive(MediaTypes.image, messageId)) {
+				cancelDownload(MediaTypes.image, messageId);
+				setLoading(false);
+				return setToDownload(true);
+			}
+
+			if (toDownload && !loading) {
+				return handleDownload();
+			}
+
 			if (!showAttachment) {
 				return;
 			}
 
-			return showAttachment(file);
+			return showAttachment(newFile);
 		};
 
-		if (file.description) {
+		if (newFile.description) {
 			return (
 				<Button disabled={isReply} theme={theme} onPress={onPress}>
 					<View>
 						<Markdown
-							msg={file.description}
+							msg={newFile.description}
 							style={[isReply && style]}
 							username={user.username}
 							getCustomEmoji={getCustomEmoji}
 							theme={theme}
 						/>
-						<MessageImage imgUri={img} theme={theme} />
+						<MessageImage imgUri={img} toDownload={toDownload} loading={loading} />
 					</View>
 				</Button>
 			);
 		}
 
 		return (
 			<Button disabled={isReply} theme={theme} onPress={onPress}>
-				<MessageImage imgUri={img} theme={theme} />
+				<>
+					<MessageImage imgUri={img} toDownload={toDownload} loading={loading} />
```

Comment:
You're caching twice with FastImage on AttachmentView.

Reply:
As I talked with you, I can't handle the download just with FastImage.preload, so inside the AttachmentView -> ImageComponent -> I check if the URL starts with LOCAL_DOCUMENT_PATH, if it's true, we use the Image from react-native